### DISCUSSION TITLE
Flat action on repository with branch protection

### DISCUSSION THREAD

**Discussion by flavianmissi** [githubocto/flat-action#66] [Q&A] [answered]
Hi,

I have a repository where the master branch is protected against direct pushes (require a pull request before merging). This means that the `flat-data` user cannot push to the master branch directly.

Is there a way to circumvent this? My first instinct is that `flat-data` should create PRs rather then push directly, but I couldn't find a way to set this up, so I assume there isn't one.

Thanks in advance!

**Comment by irealva** [comment, marked as answer]
Hi @flavianmissi you're correct we haven't really set this up unfortunately. Could you maybe create the Flat action in another branch that is unprotected then open PRs manually in master branch pulling in the file? 

Just a random suggestion, I think the actual right approach of opening PRs would require a PR to the Flat action! You're welcome to work on it and we can review.

**Reply by flavianmissi** [reply]
Thx @irealva! I'll try your suggestion for now. And hopefully I can find the time to put together a PR to the Flat action 😸
